Sources/SKCore/BuildSystemManager.swift

```diff
@@ -47,25 +54,25 @@ public final class BuildSystemManager {
 
   /// Create a BuildSystemManager that wraps the given build system. The new manager will modify the
   /// delegate of the underlying build system.
-  public init(buildSystem: BuildSystem, mainFilesProvider: MainFilesProvider?) {
-    precondition(buildSystem.delegate == nil)
+  public init(buildSystem: BuildSystem?, mainFilesProvider: MainFilesProvider?) {
```

Comment:
I hadn't thought about it before, but would it make sense for the fallback build system to also be optional? It seems useful to be able to turn it off, particularly for test code.  Otherwise it can hide bugs in tests.

Reply:
I'd also be okay merging this first and treating that as a separate improvement if you prefer; I don't think it really impacts the rest of the API).

Reply:
Done